CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Core.Entity;
+using UnityEngine;
+
+public class PlayerAbilityComponent : MonoBehaviour
+{
+    public DashAbility DashAbility { get; private set; }
+    public ShieldAbility ShieldAbility { get; private set; }
+
+    public List<PlayerAbility> playerAbilities;
+    private void Awake()
+    {
+        DashAbility = GetComponent<DashAbility>();
+        ShieldAbility = GetComponent<ShieldAbility>();
```

Comment:
same here, if you're going to assume the object has these components, you should do a requirescomponent or assert if it's missing